streams/src/main/java/org/apache/kafka/streams/processor/internals/ProcessorStateManager.java

```diff
@@ -186,7 +186,7 @@ public ProcessorStateManager(final TaskId taskId,
         this.changelogReader = changelogReader;
         this.sourcePartitions = sourcePartitions;
 
-        this.baseDir = stateDirectory.directoryForTask(taskId);
+        this.baseDir = stateDirectory.getOrCreateDirectoryForTask(taskId);
```

Comment:
Note: the only logical changes are in `StateDirectory` and `StateDirectoryTest`, the rest of the files were just touched by renaming this method to include the `getOrCreate` prefix, since this is otherwise pretty opaque.